src/Parser/Core/Modules/Features/Checklist2/PreparationRule.js

```diff
@@ -58,6 +58,66 @@ class PreparationRule extends React.PureComponent {
       </React.Fragment>
     );
   }
+  renderFlaskRequirements() {
+    const { thresholds } = this.props;
+
+    return (
+      <React.Fragment>
+        <Requirement
+          name="Flask is used"
+          thresholds={thresholds.flaskPresent}
+        />
+      </React.Fragment>
+    );
+  }
+  renderFoodRequirements() {
+    const { thresholds } = this.props;
+
+    return (
+      <React.Fragment>
+        <Requirement
+          name="Food is up"
+          thresholds={thresholds.foodPresent}
+        />
+        <Requirement
+          name="Best food is up"
+          thresholds={thresholds.bestFoodPresent}
+        />
+      </React.Fragment>
+    );
+  }
+  renderGemRequirements() {
+    const { thresholds } = this.props;
+
+    return (
+      <React.Fragment>
+        {thresholds.socketsGemmed.max !== 0 &&
```

Comment:
this may be preference (a verdict from @MartijnHols on style would be helpful), but: with a lot of conditions like this, you should instead use element variables. For example this first one would be:

```jsx
let socketsGemmed = null;
if (thresholds.socketsGemmed.max !== 0) {
    socketsGemmed = <Requirement ... />;
}
```

with a `{socketsGemmed}` in its place within the fragment.